baseplate/lib/metrics.py

```diff
@@ -62,7 +62,17 @@
 
 
 def _metric_join(*nodes: bytes) -> bytes:
-    return b".".join(node.strip(b".") for node in nodes)
+    return b".".join(node.strip(b".") for node in nodes if node)
+
+
+def _format_tags(tags: Optional[Dict]) -> Optional[str]:
+    if not tags:
+        return None
+
+    parts = [""]
+    for key, value in tags.items():
+        parts.append(f"{key}={value}")
+    return ",".join(parts)
```

Comment:
since we have to call `encode()` every time we sue this function, how bout doing the encode here to save time for callers?